tests/TestCase/View/Helper/FormHelperTest.php

```diff
@@ -3678,20 +3679,36 @@ public function testControlSelectType(): void
             EnumType::from(ArticleStatusLabelInterface::class)
         );
 
-        $article = $articlesTable->newEmptyEntity();
-        $this->Form->create($article);
+        $this->Form->create($articlesTable->newEmptyEntity());
         $result = $this->Form->control('published');
+        $this->assertHtml($expected, $result);
+
+        $articlePriosTable = $this->getTableLocator()->get('ArticlePrios');
+        $articlePriosTable->getSchema()->setColumnType(
+            'priority',
+            EnumType::from(Priority::class)
+        );
+
+        $this->Form->create($articlePriosTable->newEmptyEntity());
+        // Select empty by default
+        $result = $this->Form->control('priority', ['empty' => ['' => ' - please select - '], 'default' => '']);
```

Comment:
Is it expected that "empty" set requires also "default" to be set?
Otherwise it would not use empty, and use the DB column default instead.

Reply:
Showing the db default as selected is expected. Having an empty option when the db has a default seems contradictory to me but since you can achieve what you need by explicitly setting required options, no changed is required.

Reply:
Let's look at it from the other side:
Having a not null field being added an empty value in form select that is not being used or usable, how does that make sense?

So I would argue that if someone adds an empty value to a select control, it should serve a useful purpose.
In this case forcing the user to make a clear decision on what to select instead of "accidentally" selecting/confirming it on submit.
And in this case it would be better to not use the DB column default then IMO.

To make things consistent, we could also have the same behavior for not null fields as soon as empty value is present.

The auto-value through default for automated adding in the background can still be useful, e.g. if done through some non-human storing (and thus no form and empty value being present anyway).

I can understand if this is too much of a fringe case and that people should instead move their DB defaults to the app layer. But then, what's the benefit of having the option to store them on DB?

Also note that there should be a semantic difference of adding empty value compared to just

    array_unshift($options, $value)
    
The latter would behave in the way that the current behavior is, and how the empty behavior should not be.
It basically means to add a first empty value without changing the default behavior.

Reply:
> Is it expected that "empty" being set requires also "default" to be set?

Yes, empty and default are not directly connected. If you have a default value in the database that value  should be the default value. 

> So I would argue that if someone adds an empty value to a select control, it should serve a useful purpose.
In this case forcing the user to make a clear decision on what to select instead of "accidentally" selecting/confirming it on submit. And in this case it would be better to not use the DB column default then IMO.

Aren't you able to achieve that today by setting `empty` and `default`?

Reply:
Yes :)
I am fine with this PR being merged and done.

But I am arguing for a possible behavior change towards 5.1 etc. To a more useful and sane default behavior.

Reply:
Forcing the user to select a value when the system already has a preferred default is surely a fringe use case. I see no need for any change even in 5.1 when what you need can already be achieved by overriding the default too.

Reply:
Well, if the dev wouldnt want this, they wouldnt add the empty case in the first place :)

But meeh, I will ignore this little issue for now. Bigger fish to fry.

Shall we merge this test case as is?